food-truck-frontend/src/components/notifications/NotificationListComponent.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from "react";
+import { Grid } from "@material-ui/core";
+import NotificationComponent from "./NotificationComponent";
+import { Notification, getNotifications } from "../../api/Notification";
+import { NotificationData } from "../../redux/notifications/NotificationReducer";
+import getUserInfo from "../../util/token";
+
+export type NotificationListProps = {
+  data: NotificationData,
+  loadNotificationsFromBackend: () => Promise<void>;
+};
+
+function NotificationListComponent(props: NotificationListProps) {
+  const [notifications, setNotifications]: [Notification[], any] = useState([]);
+  const [initialized, setInitialized]: [boolean, any] = useState(false);
+  useEffect(() => {
+    var id: number | undefined = getUserInfo()?.userID;
+    if (!initialized) {
+      // Load
+      if (id !== undefined && id !== 0) {
+        props.loadNotificationsFromBackend().then();
+        setNotifications(props.data.notifications);
+      }
+      // updateNotifications().then();
+      setInitialized(true);
+    } else {
+      const timer = setInterval(() => {
+        var id: number | undefined = getUserInfo()?.userID;
+        if (id !== undefined && id !== 0) {
+          props.loadNotificationsFromBackend().then();
+          setNotifications(props.data.notifications);
```

Comment:
This kind of confuses me. Why are you storing it in both the props and the state? Couldn't you just refer to the props instead (since, with Redux, the component will re-render when its props are updated)?

Reply:
The reason we need it in the state as well is so we can delete notifications on the frontend. If we didn't do this, deleted notifications would still be present until we fetch them again from the backend.